dietpi/dietpi-software

```diff
@@ -14467,7 +14474,14 @@ _EOF_
 		Banner_Setup
 		Banner_Apt_Update
 		# - Set noninteractive | Allows automation for some software
-		DEBIAN_FRONTEND='noninteractive' apt-get upgrade -y --force-yes
+		local force_options='--force-yes'
+
+		if (( $DISTRO >= 4 )); then
+
+			force_options='--allow-downgrades --allow-remove-essential --allow-change-held-packages --allow-unauthenticated'
+
+		fi
+		DEBIAN_FRONTEND='noninteractive' apt-get upgrade -y $force_options
 
```

Comment:
@Fourdee
I thought already longer about a central APT function file (with AGI, AGU, AGP etc.), that we can source on top of all our scripts. Further we can include other helpful global variables, like $DISTRO and stuff like that there.

But for now just copying the force_options selection solves little deprecated option warning shown on dietpi-software.

Reply:
Yep, been thinking same 👍 

I'am also considering a ```global.sh``` script, that we can execute, which will export all DietPi shared globals used in scripts. But this would need looking into, as we currently export the ```.*``` results in stages during boot (eg: when they are available).
Also need to consider when this would need to be run (eg: updated), and, if updating all will increase load times etc.